provisioner/puppet-masterless/provisioner.go

```diff
@@ -62,45 +65,48 @@ type Config struct {
 
 	// If true, packer will ignore all exit-codes from a puppet run
 	IgnoreExitCodes bool `mapstructure:"ignore_exit_codes"`
-
-	// The Guest OS Type (unix or windows)
-	GuestOSType string `mapstructure:"guest_os_type"`
 }
 
 type guestOSTypeConfig struct {
+	tempDir          string
 	stagingDir       string
 	executeCommand   string
 	facterVarsFmt    string
 	facterVarsJoiner string
 	modulePathJoiner string
 }
 
+// FIXME assumes both Packer host and target are same OS
 var guestOSTypeConfigs = map[string]guestOSTypeConfig{
 	provisioner.UnixOSType: {
+		tempDir:    "/tmp",
 		stagingDir: "/tmp/packer-puppet-masterless",
 		executeCommand: "cd {{.WorkingDir}} && " +
 			`{{if ne .FacterVars ""}}{{.FacterVars}} {{end}}` +
 			"{{if .Sudo}}sudo -E {{end}}" +
 			`{{if ne .PuppetBinDir ""}}{{.PuppetBinDir}}/{{end}}` +
-			`puppet apply --verbose --modulepath='{{.ModulePath}}' ` +
```

Comment:
What's your reasoning behind removing the verbose and modulepath flags?

Reply:
I didn't remove modulepath, just wrapped it in an IF. So that if the template omitted it, the defaults would get used. Furthermore, I do things like:

``{{if ne ".ModulePath" ""}}--modulepath="{{.ModulePath}}:$({{.PuppetBinDir}}/puppet config print {{if ne "{{user `environment`}} ""}}--environment={{user `environment`}}{{end}} modulepath)'{{end}}"`

IMO '--verbose' is entirely too wordy and useful information gets lost in the deluge of output.

Reply:
Ah, now I see it. That seems like a good change for modulepath.

As for the --verbose... I have no strong opinion either way; just wanted to know your reasoning for the change. In general I tend to agree that it's probably spammy. Go ahead and keep it out.